### PR TITLE
Support count function

### PR DESCRIPTION
**Pull request by thuan1412**
Related to #89 

- [x] Write test

### CODE REVIEW

File: internal/handlers/jsonb1/msg_count.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	if ok && len(projection.Map()) != 0 {
+		return nil, common.NewErrorMessage(common.ErrNotImplemented, "MsgFind: projection is not supported")
+	}
+
+	// in count query, key of filter valyes if is "query"
+	filter, _ := m["query"].(types.Document)
+	sort, _ := m["sort"].(types.Document)
+	limit, _ := m["limit"].(int32)
+
+	sql := fmt.Sprintf(`SELECT count(_jsonb) FROM %s`, pgx.Identifier{db, collection}.Sanitize())
```
[the remark this answers is not in the document]
Comment (thuan1412) [reply internal/handlers/jsonb1/msg_count.go:51]: Yep, I will extract the duplicated code into a third function.

[the remark this answers is not in the document]
Comment (thuan1412) [reply internal/handlers/jsonb1/msg_count.go:51]: After trying to create the third function, I see that the duplication does not reduce much.
Moreover, the MsgFind and MsgCount have only four differences:
1. `MsgCount` uses `m["count"]` to get the collection name, while `MsgFind` uses `m["find"]`.
2. `MsgCount` uses `m["query"]` to get the the filter conditions, while `MsgFind` uses `m["filter"]`.
3. The `count(_jsonb)` and `_json` in the Select query.
4. `MsgCount` responds  a number, while `MsgFind` responses a list of documents.

The first three differences can be grouped into a single `if else` block.

What do you think if we put `MsgFind` and `MsgCount` into a single file? Like what you have said in [this comment](https://github.com/FerretDB/FerretDB/issues/89#issuecomment-978934912).

[the remark this answers is not in the document]
Comment (AlekSi) [reply internal/handlers/jsonb1/msg_count.go:51]: Sure, sounds great! (yep, that's your friendly @ferretdb-bot)

File: internal/handlers/jsonb1/msg_find_or_count.go [hunk 2]
```diff
@@ -27,27 +27,39 @@ import (
...
+		filter, _ = m["query"].(types.Document)
+		sql = fmt.Sprintf(`SELECT COUNT(*) FROM %s`, pgx.Identifier{db, collection}.Sanitize())
+	}
 
-	filter, _ := m["filter"].(types.Document)
 	sort, _ := m["sort"].(types.Document)
 	limit, _ := m["limit"].(int32)
 
-	sql := fmt.Sprintf(`SELECT _jsonb FROM %s`, pgx.Identifier{db, collection}.Sanitize())
+	// start the duplicate code
```
Comment (OpenSauce) [inline internal/handlers/jsonb1/msg_find_or_count.go:62]: Can remove this :)

Comment (thuan1412) [reply internal/handlers/jsonb1/msg_find_or_count.go]: Sure. I forgot to delete this line.

File: internal/handlers/handler_test.go [hunk 3]
```diff
@@ -468,3 +468,82 @@ func TestFind(t *testing.T) {
...
+func TestCount(t *testing.T) {
+	t.Parallel()
+
+	ctx := testutil.Ctx(t)
+	pool := testutil.Pool(ctx, t)
+	l := zaptest.NewLogger(t)
+	shared := shared.NewHandler(pool, "127.0.0.1:12345")
+	sql := sql.NewStorage(pool, l.Sugar())
+	jsonb1 := jsonb1.NewStorage(pool, l)
+	handler := New(pool, l, shared, sql, jsonb1)
```
Comment (AlekSi) [inline internal/handlers/handler_test.go:481]: It uses options structure now: https://github.com/FerretDB/FerretDB/blob/c52ea450072bd3453f27146c49a6869a8a7c25fb/internal/handlers/handler_test.go#L42-L49

**Comment by AlekSi** [comment]
Please ping me when it is ready for review

**Comment by thuan1412** [comment]
@AlekSi It's ready for review.

**Review by AlekSi — APPROVED** [review]

**Comment by AlekSi** [comment]
Thank you!

**Pull request merged**